pkgs/development/compilers/p4c/default.nix

```diff
@@ -0,0 +1,85 @@
+{ lib
+, stdenv
+, fetchFromGitHub
+, cmake
+, boehmgc
+, bison
+, flex
+, protobuf
+, gmp
+, boost
+, python3
+, doxygen
+, graphviz
+, libbpf
+, libllvm
+, enableDocumentation ? true
+, enableBPF ? true
+, enableDPDK ? true
+, enableBMV2 ? true
+, enableGraphBackend ? true
+, enableP4Tests ? true
+, enableGTests ? true
+, enableMultithreading ? false
+}:
+let
+  toCMakeBoolean = v: if v then "ON" else "OFF";
+in
+stdenv.mkDerivation rec {
+  pname = "p4c";
+  version = "v1.2.2.1";
+
+  src = fetchFromGitHub {
+    owner = "p4lang";
+    repo = "p4c";
+    rev = version;
+    sha256 = "sha256-XIZ7Cm5zfr5XA+s0aSG1WwWHCPjAYv/YoBWTRaXi9rQ=";
+    fetchSubmodules = true;
```

Comment:
Please remove the unused modules.

Reply:
Do you mean with `sparseCheckout` ?

Reply:
More like removing it in postFetch

Reply:
Hm, alright, but, what is the objective here?

Reply:
Done.